browser/brave_content_browser_client.cc

```diff
@@ -1358,16 +1358,20 @@ bool PreventDarkModeFingerprinting(WebContents* web_contents,
 
 bool BraveContentBrowserClient::OverrideWebPreferencesAfterNavigation(
     WebContents* web_contents,
+    content::SiteInstance& main_frame_site,
     WebPreferences* prefs) {
   bool changed =
       ChromeContentBrowserClient::OverrideWebPreferencesAfterNavigation(
-          web_contents, prefs);
+          web_contents, main_frame_site, prefs);
   return PreventDarkModeFingerprinting(web_contents, prefs) || changed;
 }
 
-void BraveContentBrowserClient::OverrideWebkitPrefs(WebContents* web_contents,
-                                                    WebPreferences* web_prefs) {
-  ChromeContentBrowserClient::OverrideWebkitPrefs(web_contents, web_prefs);
+void BraveContentBrowserClient::OverrideWebPreferences(
+    WebContents* web_contents,
+    content::SiteInstance& main_frame_site,
```

Comment:
I think we also need to change `PreventDarkModeFingerprinting` to use `main_frame_site` instead of `web_contents->GetLastCommittedURL()` (which I think was not right and should have been `web_contents->GetPrimaryMainFrame()->GetSiteInstance()->GetSiteURL()`)

Reply:
Fixed here https://github.com/brave/brave-core/pull/27172/commits/8571276ce9fd7248970258df8fc90179a3a69aa6